core/src/main/java/nl/nn/adapterframework/receivers/ExchangeMailListener.java

```diff
@@ -99,28 +98,28 @@ public Message extractMessage(Item rawMessage, Map<String,Object> threadContext)
 		if (!EMAIL_MESSAGE_TYPE.equals(getMessageType())) {
 			return super.extractMessage(rawMessage, threadContext);
 		}
-		Item item = (Item) rawMessage;
+
 		try {
 			XmlBuilder emailXml = new XmlBuilder("email");
 			EmailMessage emailMessage;
 			PropertySet ps;
 			if (isSimple()) {
 				ps = new PropertySet(EmailMessageSchema.Subject);
-				emailMessage = EmailMessage.bind(getFileSystem().getExchangeService(), item.getId(), ps);
+				emailMessage = EmailMessage.bind(getFileSystem().getExchangeService(), rawMessage.getId(), ps);
 				emailMessage.load();
 				addEmailInfoSimple(emailMessage, emailXml);
 			} else {
 				ps = new PropertySet(EmailMessageSchema.DateTimeReceived, EmailMessageSchema.From, EmailMessageSchema.Subject, EmailMessageSchema.Body, EmailMessageSchema.DateTimeSent);
-				emailMessage = EmailMessage.bind(getFileSystem().getExchangeService(), item.getId(), ps);
+				emailMessage = EmailMessage.bind(getFileSystem().getExchangeService(), rawMessage.getId(), ps);
 				emailMessage.load();
 				addEmailInfo(emailMessage, emailXml);
 			}
 
-			if (StringUtils.isNotEmpty(getStoreEmailAsStreamInSessionKey())) {
+			if (StringUtils.isNotEmpty(getStoreEmailInSessionKey())) {
 				emailMessage.load(new PropertySet(ItemSchema.MimeContent));
 				MimeContent mc = emailMessage.getMimeContent();
-				ByteArrayInputStream bis = new ByteArrayInputStream(mc.getContent());
-				threadContext.put(getStoreEmailAsStreamInSessionKey(), bis);
+				Message message = new Message(mc.getContent(), mc.getCharacterSet());
+				threadContext.put(getStoreEmailInSessionKey(), message);
```

Comment:
Je stopt hier een Message in de sessie, in plaats van een byte array. Het is maar de vraag of de ontvangende partij daarmee om kan gaan.